src/aap_eda/tasks/analytics.py

```diff
@@ -32,22 +33,36 @@ def schedule_gather_analytics(queue_name: str = ANALYTICS_TASKS_QUEUE) -> None:
         return
     interval = utils.get_analytics_interval()
     logger.info(f"Schedule analytics to run in {interval} seconds")
-    tasking.enqueue_delay(
-        queue_name,
-        ANALYTICS_SCHEDULE_JOB_ID,
-        interval,
-        auto_gather_analytics,
-    )
+    with advisory_lock(ANALYTICS_SCHEDULE_JOB_ID, wait=False) as acquired:
+        if not acquired:
+            logger.debug(
+                "Another instance of schedule_gather_analytics"
+                " is already running",
+            )
+            return
 
+        tasking.enqueue_delay(
+            queue_name,
+            ANALYTICS_SCHEDULE_JOB_ID,
+            interval,
+            auto_gather_analytics,
+        )
 
-def auto_gather_analytics(queue_name: str = ANALYTICS_TASKS_QUEUE) -> None:
-    gather_analytics(queue_name)
+
+def auto_gather_analytics() -> None:
     schedule_gather_analytics()
+    gather_analytics()
```

Comment:
@bzwei @hsong-rh Note here that I have changed the command because there is no more enqueueing of the function. All is executed each time, so first we schedule the next execution of the gatherer and later run it, otherwise if gather_analytics fails for some reason, the next execution would not be scheduled.